services/web-server/src/login/strategies/github.js

```diff
@@ -22,12 +22,22 @@ module.exports = class Github {
     Object.assign(this, strategyCfg);
 
     this.identityProviderId = 'github';
-    this.githubClient = new GithubClient();
+    this.GithubAccessToken = GithubAccessToken;
+    this.clients = new Map();
   }
 
   async getUser({ username, userId }) {
     const user = new User();
-    const [githubErr, githubUser] = await tryCatch(this.githubClient.userFromUsername(username));
+    const githubEntry = await this.GithubAccessToken.load({ userId: String(userId) }, true);
+
+    if (!githubEntry) {
+      debug(`Github user id ${userId} could not be found in the database.`);
+
+      return;
+    }
+
+    const githubClient = this.clients.get(userId);
```

Comment:
What happens if this isn't set?

Reply:
Good catch. This is no bueno. We should make sure it's set.